src/tetris/scene/playScene.ts

```diff
@@ -137,53 +92,111 @@ export default class PlayScene extends Phaser.Scene {
 	private _pauseButton: TextButton;
 	private _scoreWidget: ScoreWidget;
 	private _countdownWidget: CountdownWidget;
 	private _scoreboardWidget: ScoreboardWidget;
 	private readonly _game: Game;
-	private _pipeline: Phaser.Renderer.WebGL.Pipelines.TextureTintPipeline;
+	private _rainbowPipeline: Phaser.Renderer.WebGL.Pipelines.TextureTintPipeline;
+	private _startTimerStarted: boolean;
+	private _localSocketId: String;
+	private _backgroundGraphicsPipeline: Phaser.Renderer.WebGL.Pipelines.TextureTintPipeline;
+	private _backgroundSprite: Phaser.GameObjects.Sprite;
 	//endregion
-
+	
 	//region private methods
-	private _setupNetworkingClient(): void {
-		this._game.networkingClient.receive("playerLeft", (args) => {
-			console.log('playerLeft:', args.id);
-			this._removeRemoteField(args.id);
-		});
-		this._game.networkingClient.receive("playerJoined", (args) => {
-			console.log('playerJoined:', args.id);
-			this._addRemoteField(args.id);
+	private _createBackground(): void {
+		this._backgroundGraphicsPipeline = new Phaser.Renderer.WebGL.Pipelines.TextureTintPipeline({
+			game: this._game,
+			renderer: this._game.renderer, 
+			fragShader: this.cache.shader.get('interstellar')
 		});
-		this._game.networkingClient.receive("fieldUpdate", (args) => {
-			const field = this._remotePlayerFields.get(args.id);
-			if (!field) {
-				console.warn("WARNING: Had to create player field; currentplayers MISSED?");
-				this._addRemoteField(args.id);
+		(this._game.renderer as Phaser.Renderer.WebGL.WebGLRenderer).addPipeline('interstellar', this._backgroundGraphicsPipeline);
+
+		this._backgroundGraphicsPipeline.setFloat2('uResolution', config.graphics.width, config.graphics.height);
+		this._backgroundGraphicsPipeline.setFloat1('uTime', 0.0);
+		this.adjustBackgroundParameters(0.10, [0.2, 0.2, 0.2]);
+
+		this._backgroundSprite = this.add.sprite(config.graphics.width / 2, config.graphics.height / 2, 'noise');
+		const scaleX = config.graphics.width / 256.0;
+		const scaleY = config.graphics.height / 256.0;
+		this._backgroundSprite.setScale(scaleX, scaleY);
+		
+		this._backgroundSprite.setPipeline('interstellar');
+	}
+
+	private _registerNetworkEvents(): void {
+		this._game.networkingClient.receive("matchUpdate", this._updateMatch.bind(this));
+	}
+
+	private _startMatch(): void {
+		if (this._localPlayerField.fieldState === FieldState.Loss) {
+			return;
+		}
+		console.log('starting match');
+		this.adjustBackgroundParameters(0.25, [0.7, 0.7, 0.7]);
+		this._localPlayerField.reset();
+	}
+
+	private _updateMatch(match: Match) {
+		console.log(match);
+		this._match = match;
+		const matchHasStarted = Date.now() > Date.parse(match.startTime);
+		if (!this._startTimerStarted && !matchHasStarted) {
+			this._startTimerStarted = true;
+			setTimeout(this._startMatch.bind(this), Date.parse(match.startTime) - Date.now(), {});
+		}
+		for (const [index, player] of match.players.entries()) {
+			if (player.socketId === this._localSocketId) {
```

Comment:
Nesting-level and readability.
I would suggest to split it up to multiple private functions.
That would increase readability as well.

Reply:
I just refactored this, would you have another look, please?